### DISCUSSION TITLE
Passes with different groupingIdentifier value are grouped inside the wallet app

### DISCUSSION THREAD

**Discussion by cuni0716** [alexandercerutti/passkit-generator#198] [Q&A]
Hello again @alexandercerutti!
Sorry for being a pain, I'm just finishing my implementation so it will be my last question for you (I hope 🤞).

The thing here is that I create different passes with different `groupingIdentifier` value but those passes are grouped once installed on the device.

Have you any idea why?

I attach two pass.json files for you to check and please, let me know if you need further info.
[1.json](https://github.com/alexandercerutti/passkit-generator/files/14943275/1.json)
[2.json](https://github.com/alexandercerutti/passkit-generator/files/14943276/2.json)

Thanks again and again for the project and for your help, have a nice day!

**Comment by alexandercerutti** [comment]
Hey @cuni0716, sorry for the delay! I could not respond earlier.

Apple Wallet documentation for `groupingIdentifier` reports this, and I cannot find anything else in the archive documentation:

> An identifier the system uses to group related boarding passes or event tickets. Wallet displays passes with the same groupingIdentifier, passTypeIdentifier, and type as a group.
>
> Use this identifier to group passes that are tightly related, such as boarding passes for different connections on the same trip.

So I think they are getting grouped for the other two fields...

**Comment by cuni0716** [comment]
Yes, that's the same that I found. Bad news for me but I will investigate a bit more.

Do you know how can I use a new PKPass instance each time I use `PKPass.from`?

I have a problem with the translations and I think this way I would solve it.

Again thanks a lot @alexandercerutti !!

**Reply by alexandercerutti** [reply]
You can pass a `PkPass` instance to `PkPass.from` to clone all the things

**Reply by cuni0716** [reply]
Yes, but what I need is the opposite, a brand new instance each time. How can I do that?

**Reply by cuni0716** [reply]
Anyway, I will close this one since I examined the content of the .pkpass files and the groupIdentifier are correctly populated by your package, so not an issue for your side.

**Reply by alexandercerutti** [reply]
So I perhaps misunderstood your phrase:

> Do you know how can I use a new PKPass instance each time I use PKPass.from

What did you mean?
Every time you use `new PKPass` or `PKPass.from`, you are creating a new instance

**Reply by cuni0716** [reply]
The second time in a row that I call PKPass.from method it already have the localize files from the first time. I expect a brand new instance without any file that has not been added explocitly. That was my comment about.

**Reply by alexandercerutti** [reply]
If you use the same model, the same data should get loaded. If, by calling two times `PKPass.from`, you find on the second instance some files you added to the first `PkPass.from` via `addBuffers`, that might be a bug...
